File: app/models/character.rb
# frozen_string_literal: true

class Character < ApplicationRecord
  validates :name, :height, :mass, :hair_color, :skin_color, :eye_color, :birth_year, :gender, presence: true
  validates :height, :mass, numericality: { greater_than_or_equal_to: 0 }

  belongs_to :Planet

  has_many :characters_species
  has_many :species, through: :characters_species

  has_many :characters_films
  has_many :films, through: :characters_films

  has_many :characters_foods
  has_many :foods, through: :characters_foods
end

File: app/javascript/components/Character.js
import React from "react"
import PropTypes from "prop-types"
import {Link} from 'react-router-dom'

function Character(props) {
    return (
      <tr>
        <td><Link to={`/Characters/${props.character.id}`}>{props.character.name}</Link></td>
        <td>{props.character.height}</td>
        <td>{props.character.mass}</td>
        <td>{props.character.hair_color}</td>
        <td>{props.character.skin_color}</td>
        <td>{props.character.eye_color}</td>
        <td>{props.character.birth_year}</td>
        <td>{props.character.gender}</td>
      </tr>
    )

}

Character.propTypes = {
  character: PropTypes.object
};

export default Character
